Assets/Scripts/Models/HealthSystem.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+
+[Serializable]
+public class HealthSystem
+{
+    //all of this data should probably be generated at runtime or loaded from a file
+    public float Value
+    {
+        get
+        {
+            return _currentHealth;
+        }
+
+        private set
+        {
+            if (Invincible == true) return;
+            //We have been Revived!
+            if (_currentHealth == 0 && value > _currentHealth)
+            {
+                _currentHealth = value;
+
+                if (OnRevive != null)
+                    OnRevive();
+                return;
+            }
+            //we have been hit but we are not dead!
+            if (value < _currentHealth && _currentHealth > 0)
```

Comment:
To be honest, after testing both mine and your condition for being revived, they both seem to work the same but i think that yours is a bit more clear then mine so if you do not mind i would like to use your change to the condition.